File: MojStomatolog/UI/mojstomatolog_desktop/lib/models/product.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mojstomatolog_desktop/models/product_category.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  int? productId;
  String? name;
  String? description;
  int? productCategoryId;
  double? price;
  String? imageUrl;
  bool? active;

  ProductCategory? productCategory;

  Product();

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);
  Map<String, dynamic> toJson() => _$ProductToJson(this);
}

File: MojStomatolog/MojStomatolog.Models/Core/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MojStomatolog.Models.Core
{
    public class Product
    {
        public int ProductId { get; set; }

        public string Name { get; set; } = null!;

        public string Description { get; set; } = null!;

        public int ProductCategoryId { get; set; }

        public double Price { get; set; }

        public string ImageUrl { get; set; } = null!;

        public bool Active { get; set; }


        [ForeignKey("ProductCategoryId")]
        public ProductCategory? ProductCategory { get; set; }
    }
}